### PR TITLE
Make sidetable optional when posting subscriber

### PR DESCRIPTION
**Pull request by akhiln**
Needed for autos integration.

### CODE REVIEW

**Comment by akhiln** [comment]
@wesdotcool

File: lib/yesmail/subscriber.rb [hunk 1]
```diff
@@ -86,11 +86,11 @@ def api_remove
     # This will create all of the json from the data placed in this subscriber
     # and send it in the form of JSON to Yesmail's subscribe and send
     # composite API
-    def api_create_and_send(master, side_table)
+    def api_create_and_send(master, side_table = nil)
```
Comment (wesdotcool) [inline lib/yesmail/subscriber.rb:89]: This method use's Yesmail's SubscribeAndSendComposite API. It looks like you just want to use their Subscriber API. I would create a new method for that.

File: lib/yesmail/subscriber.rb [hunk 2]
```diff
@@ -86,11 +86,11 @@ def api_remove
...
       data = { subscriber: make_hash }
       data[:subscriberMessage] =  master.subscriber_message_data
       data[:subscriber][:allowResubscribe] = true
-      data[:sideTable] = side_table.payload_hash
+      data[:sideTable] = side_table.payload_hash unless side_table.nil?
 
       path = '/composite/subscribeAndSend'
```
Comment (wesdotcool) [inline lib/yesmail/subscriber.rb:95]: This URI will need to be changed if you're using the Subscriber API

**Comment by akhiln** [comment]
I still want to use the SubscribeAndSendComposite API. sideTable isn't a mandatory field for this API, and all relevant information for the email is attached to the subscriber for my use case.

**Comment by wesdotcool** [comment]
Well, if that's the case, then LGTM

**Comment by otherjon** [comment]
I also need side table to be optional, so since this is reviewed by Wes (LGTM also), I'll just merge it.  Warning: I'm about to submit some more pull requests that affect this file.  I intend to be fully backward-compatible, but keep your eyes open.

**Pull request merged**